core/src/ast/types.rs

```diff
@@ -113,22 +115,14 @@ impl TypeName {
                     }]),
                 },
             }),
-            TypeName::Writeable => syn::Type::Path(TypePath {
-                qself: None,
-                path: Path {
-                    leading_colon: None,
-                    segments: Punctuated::from_iter(vec![
-                        PathSegment {
-                            ident: Ident::new("diplomat_runtime", Span::call_site()),
-                            arguments: PathArguments::None,
-                        },
-                        PathSegment {
-                            ident: Ident::new("DiplomatWriteable", Span::call_site()),
-                            arguments: PathArguments::None,
-                        },
-                    ]),
-                },
-            }),
+            TypeName::Writeable => syn::parse2(quote! {
```

Comment:
suggestion (nb): https://docs.rs/syn/1.0.73/syn/macro.parse_quote.html maybe?

Reply:
I actually am using the old pattern in a lot of places, so will follow up on this in a separate PR.

Reply:
Works for me!